docs/daisy-reusing-workflows.md

```diff
@@ -0,0 +1,90 @@
+# Reusing Workflows
+
+Instead of requiring you to copy-paste large sections of workflows to perform
+similar tasks, Daisy provides two different step types for workflow reuse:
+`IncludeWorkflow` and `SubWorkflow`. Both step types will cause Daisy to run the
+workflow they specify, but how Daisy does this is different:
+
+### IncludeWorkflow
+
+`IncludeWorkflow` will run the steps of the included workflow in parallel with
```

Comment:
I'm pretty sure IncludeWorkflows run as a single step (like how you described SubWorkflow), but @adjackura or @zmarano might know better than me. By the way, both a SubWorkflow step and an IncludeWorkflow step run in parallel to other steps with which they have no dependency relation.

The biggest (probably only) difference is that IncludeWorkflows share the same resource namespace as the parent workflow. SubWorkflows do not.

Reply:
For sure, any and all steps will be run in parallel if there are no dependencies. But given that, I was talking with Andrew yesterday about this, and he explained to me (though I may have misunderstood) that `IncludeWorkflow` and `SubWorkflow` steps are considered "done" (for the purposes of dependencies) at different times: `IncludeWorkflow` immediately once all the tasks have been included (as soon as the dependency tree is rebuilt to fit the newly included workflow), and `SubWorkflow` only once all of that workflow's tasks have actually finished.

@adjackura , can you confirm?

@zmarano also commented on this same paragraph, calling for more clarity on how dependencies affect when tasks are run. I'll take another pass at it and see if I can't make it better.